File: leetcode/lc-us/cpp/1137.cpp
class Solution {
public:
    int tribonacci(int n) {
        if (n <= 1) return n;
        if (n <= 3) return n - 1;
        int first = 1, second = 1, third = 2, fourth = 4;
        for (int i = 4; i <= n; i++) {
            fourth = third + first + second;
            first = second;
            second = third;
            third = fourth;
        }
        return fourth;
    }
};

File: leetcode/lc-us/py/1137.py
class Solution:
    def tribonacci(self, n: int) -> int:
        if n <= 1:
            return n
        if n == 2:
            return 1
        a, b, c = 0, 1, 1
        for i in range(3, n + 1):
            t = c
            c = a + b + c
            a = b
            b = t
        return c
